src/api/app/jobs/update_released_binaries_job.rb

```diff
@@ -36,109 +36,107 @@ def perform(event_id)
+    #   - A BinaryRelease that is a medium has the attribute `ismedium: 'openSUSE_Leap'`.
+    # We use this hash to build the BinaryRelease.on_medium relationship from the backend data
+    medium_hash = {}
+
+    # FIXME: This job is the only thing that ever changes the BinaryRelease table. It runs in it's own queue
+    #        with one worker. Only one UpdateReleasedBinariesJob will run at once. Why do we need this transaction?
     BinaryRelease.transaction do
+      # Populate the old_binary_releases hash
       repository.binary_releases.current.unchanged.find_each do |binary|
-        key = hashkey_old_binary_releases(binary)
+        key = hashkey_binary_release(binary)
         old_binary_releases[key] = binary.slice(:disturl, :supportstatus, :binaryid, :buildtime, :id)
       end
 
-      processed_item = {}
+      new_binary_releases.each do |backend_binary|
+        backend_binary = backend_binary.with_indifferent_access
+        new_binary_release = repository.binary_releases.new(backend_binary.slice(:name, :binaryid, :binaryarch,
+                                                                                 :version, :release, :epoch,
+                                                                                 :medium, :cpeid,
+                                                                                 :disturl, :supportstatus,
+                                                                                 :updateinfoid, :updateinfoversion))
 
-      # when we have a medium providing further entries
-      medium_hash = {}
+        # `BinaryRelease`` expects a `DateTime` as attribute, the backend gives us epoch
+        new_binary_release.buildtime = Time.zone.at(backend_binary['buildtime'].to_i) if backend_binary['buildtime'].present?
 
-      new_binary_releases.each do |backend_binary|
-        # identifier
-        binary_release = { binary_name: backend_binary['name'],
-                           binary_version: backend_binary['version'] || 0, # docker containers have no version
-                           binary_release: backend_binary['release'] || 0,
-                           binary_epoch: backend_binary['epoch'],
-                           binary_arch: backend_binary['binaryarch'],
-                           medium: backend_binary['medium'],
-                           on_medium: medium_hash[backend_binary['medium']],
-                           obsolete_time: nil,
-                           modify_time: nil }
-
-        # getting activerecord object from hash, dup to unfreeze it
-        old_binary_release = old_binary_releases[hashkey_new_binary_releases(backend_binary, backend_binary['medium'])]
+        # Set releasetime to whenever the Event::Packtrack happened that triggered this job
+        new_binary_release.binary_releasetime = time
+
+        # Set defaults for versions/release
+        # FIXME: This could also happen in the database instead...
```

Comment:
we don't want to modify old database entries if possible.

Reply:
Not necessary as the default in the code is there since we create those entries...